internal/handlers/common/aggregations/aggregations.go

```diff
@@ -24,6 +24,31 @@ import (
 
 // Stage is a common interface for all aggregation stages.
 type Stage interface {
+	// FirstStage fetches document iterator, it is only used if this is
+	// the first stage of the pipeline.
+	// The returned document iterator may be originated from querying database or
+	// from in-memory value iterator.
+	// This allows first stage of the pipeline to decide how to create initial document iterator.
+	FirstStage(ctx context.Context, closer *iterator.MultiCloser) (types.DocumentsIterator, error)
+
 	// Process applies an aggregate stage on documents from iterator.
 	Process(ctx context.Context, iter types.DocumentsIterator, closer *iterator.MultiCloser) (types.DocumentsIterator, error)
 }
+
+// Aggregation is a common interface for fetching from database.
```

Comment:
Independently of everything else, that's a bad name given this aggregation does not aggregate anything – it returns original data to be processed and aggregated, and that's pretty much the opposite.